src/importer/sources/retroarch_source.py

```diff
@@ -0,0 +1,135 @@
+                    encoding="utf-8",
+                ) as open_file:
+                    playlist_json = json.load(open_file)
+            except (JSONDecodeError, OSError, KeyError):
+                logging.warning("Cannot read playlist file: %s", str(playlist_file))
+                continue
+
+            for item in playlist_json["items"]:
+                # Select the core.
+                # Try the content's core first, then the playlist's default core.
+                # If none can be used, warn the user and continue.
+                for core_path in (
+                    item["core_path"],
+                    playlist_json["default_core_path"],
+                ):
+                    if core_path not in ("DETECT", ""):
+                        break
+                else:
+                    logging.warning("Cannot find core for: %s", str(item["path"]))
+                    bad_playlists.add(playlist_file.stem)
+                    continue
+
+                # Build game
+                game_id = hashlib.md5(item["path"].encode("utf-8")).hexdigest()
+
+                values = {
+                    "source": self.source.id,
+                    "added": added_time,
+                    "name": item["label"],
+                    "game_id": self.source.game_id_format.format(game_id=game_id),
+                    "executable": self.source.executable_format.format(
+                        rom_path=item["path"],
+                        core_path=core_path,
+                    ),
+                }
+
+                game = Game(values)
+                additional_data = {}
+
+                # Get boxart
+                boxart_image_name = item["label"] + ".png"
+                boxart_image_name = boxart_image_name.replace("&", "_")
+                boxart_folder_name = playlist_file.name.split(".", 1)[0]
+                image_path = (
+                    self.source.config_location["thumbnails"]
+                    / boxart_folder_name
+                    / "Named_Boxarts"
+                    / boxart_image_name
+                )
+                additional_data = {"local_image_path": image_path}
+
+                yield (game, additional_data)
+
+        if bad_playlists:
+            raise FriendlyError(
+                _("No RetroArch Core Selected"),
+                # The variable is a newline separated list of playlists
+                _("The following playlists have no default core:")
+                + "\n\n{}\n\n".format("\n".join(bad_playlists))
+                + _("Games with no core selected were not imported"),
+            )
+
+
+class RetroarchSource(Source):
+    name = _("RetroArch")
+    available_on = {"linux"}
+    iterator_class = RetroarchSourceIterator
+
+    config_location = Location(
+        schema_key="retroarch-location",
+        candidates=(
+            shared.flatpak_dir / "org.libretro.RetroArch" / "config" / "retroarch",
```

Comment:
Couldn't we also add the windows candidate and also make the source available on windows?